jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/jdbc/SchemaAdapterNotes.java

```diff
@@ -163,15 +163,15 @@ public static String serialize(SchemaAdapterNotes notes) {
         return builder.build().toString();
     }
 
-    public static SchemaAdapterNotes deserialize(String adapterNotes, String schemaName) {
+    public static SchemaAdapterNotes deserialize(String adapterNotes, String schemaName) throws AdapterException {
         if (adapterNotes == null || adapterNotes.isEmpty()) {
-            throw new RuntimeException(getException(schemaName));
+            throw new AdapterException("Adapter notes for schema "+schemaName+" are empty or null. Please refresh the virtual schema");
         }
         JsonObject root;
         try {
             root = JsonHelper.getJsonObject(adapterNotes);
         } catch (Exception ex) {
-            throw new RuntimeException(getException(schemaName));
+            throw new AdapterException("Can not get the json object for adapter notes of schema "+schemaName+". Please refresh the virtual schema");
```

Comment:
I like the idea of making the exceptions more specific. A small change:
"Could not parse the json which is expected to be stored in the adapter notes of schema ..."